### ISSUE TITLE
Convert to use `pyproject.toml`

### ISSUE THREAD

**Issue by esc** [numba/llvmlite#970]
---

## Feature request

Convert llvmlite to use a `pyproject.toml` instead of a `setup.py`.

**Comment by esc** [comment]
As per https://github.com/numba/numba/wiki/Minutes_2023_11_07 -- I will remove this from the 0.42.0-rc1 milestone as it is no longer planned.

**Comment by detrin** [comment]
I would like to see this happen in the future. In 2025 setup.py is obsolete and rather archaic

**Comment by esc** [comment]
@detrin I agree, it would be absolutely lovely if someone could find time for this. Sadly, nothing is concretely broken, so there is no immediate need to fix this.

**Comment by detrin** [comment]
Okay good to know. If there is an update ping me. Possibly when I have time over following months. I might look into it myself.

**Comment by llimeht** [comment]
versioneer does support putting the config in pyproject.toml; it also supports being added as a build-requires rather than being versioned, which would be nice to do at the same time.

https://github.com/python-versioneer/python-versioneer/blob/master/INSTALL.md#installation-instructions
https://github.com/python-versioneer/python-versioneer?tab=readme-ov-file#build-time-dependency-mode

Of course you might also want to consider switching to setuptools_scm since that looks to be more actively maintained at this stage.

**Comment by esc** [comment]
The CI overhaul is now largely complete and this issue could potentially be solved again. However, the current solution still works so there is no immediate need to fix anything. But it would be good to move away from `setup.py` for future compatibility.
